homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,271 @@
+        if new_state is None:
+            for value in self._covers.values():
+                value.discard(entity_id)
+            for value in self._tilts.values():
+                value.discard(entity_id)
+                tilt = True if value else False
+            self._tilt = tilt
+            return
+
+        if old_state is None:
+            features = new_state.attributes[ATTR_SUPPORTED_FEATURES]
+            if features & (SUPPORT_OPEN | SUPPORT_CLOSE):
+                self._covers[KEY_OPEN_CLOSE].add(entity_id)
+            if features & (SUPPORT_STOP):
+                self._covers[KEY_STOP].add(entity_id)
+            if features & (SUPPORT_SET_POSITION):
+                self._covers[KEY_POSITION].add(entity_id)
+            if features & (SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT):
+                self._tilts[KEY_OPEN_CLOSE].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_STOP_TILT):
+                self._tilts[KEY_STOP].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_SET_TILT_POSITION):
+                self._tilts[KEY_POSITION].add(entity_id)
+                self._tilt = True
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def state_change_listener(entity, old_state, new_state):
+            """Handle cover state changes and update state."""
+            self.update_supported_features(entity, old_state, new_state)
+            self.async_schedule_update_ha_state(True)
+
+        for entity_id in self._entities:
+            new_state = self.hass.states.get(entity_id)
+            self.update_supported_features(entity_id, None, new_state)
+        self.async_schedule_update_ha_state(True)
+        async_track_state_change(self.hass, self._entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
+
+    @property
+    def should_poll(self):
+        """Disable polling for cover group."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Flag supported features for the cover."""
+        return self._supported_features
+
+    @property
+    def is_closed(self):
+        """Return if all covers in group are closed."""
+        return self._is_closed
+
+    @property
+    def current_cover_position(self):
+        """Return current position for all covers."""
+        return self._cover_position
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current tilt position for all covers."""
+        return self._tilt_position
+
+    async def async_open_cover(self, **kwargs):
+        """Move the covers up."""
+        self.hass.async_add_job(open_cover, self.hass,
```

Comment:
2 things wrong here:
 - Don't use sync methods, you can call services asynchronously.
 - You need to call the service with blocking=True, as the opening has to be done when this method is done calling.

Reply:
See the group light for an example